targets/nintendo-switch.json

```diff
@@ -0,0 +1,35 @@
+{
+  "llvm-target": "aarch64",
+  "build-tags": ["nintendoswitch", "arm64"],
+  "goos": "linux",
+  "goarch": "arm64",
+  "compiler": "aarch64-none-elf-gcc",
+  "linker": "devKitPro",
```

Comment:
Wasn't it possible to link using `lld`?

Reply:
Also, if you can use `aarch64-none-elf-gcc`, would `clang` (with the right options) work? The less dependencies, the better.

Reply:
It wasn't. LLD can't link stuff pre-compiled with GCC with custom linker script :( - It gave several errors. Also it does need DevKitPro because of the libc required for all nintendo switch deps (the DevKitPro is a modified picolibc).

We might eventually want to port devKitPro picolibc to tinygo and replace the current one, because the picolibc from devKitPro is more generic than the one we have on tinygo today. I checked if it was easy to port it, but there are some things that require changes on other tinygo targets, so I think its better a separated PR for that eventually.

Reply:
Okay, now I understand the situation better.

The `linker` field here really refers to an actual command. I think instead of repurposing the `linker` field, you should do something like this:

```json
    "libc": "libnx",
```

That is actually the field intended for changing the libc, which is the main reason all these custom build steps are needed. You can then check for libnx (similar to picolibc) and if it is used, add the correct paths by querying the `DEVKITPRO` environment variable. Not how I would hope it would work, but I think that is the cleanest way to do it.

Reply:
On second thought, you need to run a command in the `$DEVKITPRO` directory. Which is not known statically. This requires some logic to determine the correct path.

This is somewhat of a mismatch between how I intended TinyGo to work and how DevKitPro works. I'm not sure how to fix that exactly. Maybe as it is right now.

Reply:
> Maybe as it is right now.

@aykevl do you mean as this PR has already implemented?

Reply:
Yes. I don't like it but I don't see a better way, so maybe just do it as the PR is now. It's unfortunate that devkitPro assumes the use of global variables, which isn't very compatible with TinyGo right now.